README.md

```diff
@@ -91,6 +91,53 @@ To clean the project:
 
     make clean
 
+## Limitations
+
+We should note some known limitations and implementation divergences regarding the [main siva reference specification](https://github.com/src-d/go-siva/blob/master/SPEC.md)
+
+All the issues commented below are related to the `index` part of the blocks since there is where siva really place the metadata. Almost all this meta information is encoded as unsigned values, because of that, most of the problems come from the lack of unsigned values in the `JVM`.
+
+To avoid these limitations, in some cases, a cast to a bigger number type and a binary `AND` operation with a mask solves the problem. The trick consists of:
+
+```
+unsigned int8 (byte in Go): 255
+
+if you read this byte in java, it interprets the value as signed so the same bits in java result on:
+
+signed int8 (byte in Java): -1
+
+Casting this value to a java integer, keeps the value as  -1, so we apply a binary mask, with the less weight byte set to all "ones" and the rest of the byte to "zeros":
+
+byte b = readByte() // 255 read, but in java the value is -1
+int mask = 0x000000FF
+int n = b & mask // now n is an integer storing the value 255
+
+```
+
+This procedure is related on how `JVM` encodes the number values using [two's complement](https://en.wikipedia.org/wiki/Two%27s_complement) and it can apply for all the types which can be cast to a bigger number type.
+
+***Unsigned Integer 64 Limitation!!!***: a siva file with a value in those fields that the specification encodes as `uint64 ` can contain values in range [0, 2<sup>64</sup>-1] while java implementation only supports values in range [0, 2<sup>64-1</sup>-1]. There's no a number type bigger than a `long` (int64) in java, so this can't be avoided.
```

Comment:
I think one "!" is enough :stuck_out_tongue:

Reply:
Actually, there's `BigInteger`, which I think has a constructor accepting a byte array. Can't we used that to emulate `uint64`?

(But perhaps is slow enough to not be worth it)

Reply:
The problem is that these uint64 values will be use then to manipulate the underlaying siva file, so eventually they have to be converted to `long` types again because there's no methods to index and map files that accept `BigInteger` types.

Reply:
Fair enough